stripe/src/main/java/com/stripe/android/model/PaymentIntent.java

```diff
@@ -0,0 +1,299 @@
+        return mConfirmationMethod;
+    }
+
+    public long getCreated() {
+        return mCreated;
+    }
+
+    public String getCurrency() {
+        return mCurrency;
+    }
+
+    public String getDescription() {
+        return mDescription;
+    }
+
+    public boolean isLiveMode() {
+        return mLiveMode;
+    }
+
+    public Map<String, Object> getNextSourceAction() {
+        return mNextSourceAction;
+    }
+
+    public Uri getAuthorizationUrl() {
+        if (mStatus.equals("requires_source_action") &&
+                mNextSourceAction.containsKey("type") &&
+                mNextSourceAction.get("type").equals("authorize_with_url") &&
+                mNextSourceAction.containsKey("value") && mNextSourceAction.get("value") instanceof Map) {
+            Map<String, String> urlMap = (Map) mNextSourceAction.get("value");
+            if (urlMap.containsKey("url")) {
+                return Uri.parse(urlMap.get("url"));
+            }
+        }
+        return null;
+    }
+
+    public String getReceiptEmail() {
+        return mReceiptEmail;
+    }
+
+    public ShippingMethod getShippingMethod() {
+        return mShippingMethod;
+    }
+
+    public String getSource() {
+        return mSource;
+    }
+
+    public String getStatus() {
+        return mStatus;
+    }
+
+    PaymentIntent(
+            String id,
+            String objectType,
+            List<String> allowedSourceTypes,
+            Long amount,
+            Long canceledAt,
+            String captureMethod,
+            String clientSecret,
+            String confirmationMethod,
+            Long created,
+            String currency,
+            String description,
+            Boolean liveMode,
+            Map<String, Object> nextSourceAction,
+            String receiptEmail,
+            ShippingMethod shippingMethod,
+            String source,
+            String status
+    ) {
+        mId = id;
+        mObjectType = objectType;
+        mAllowedSourceTypes = allowedSourceTypes;
+        mAmount = amount;
+        mCanceledAt = canceledAt;
+        mCaptureMethod = captureMethod;
+        mClientSecret = clientSecret;
+        mConfirmationMethod = confirmationMethod;
+        mCreated = created;
+        mCurrency = currency;
+        mDescription = description;
+        mLiveMode = liveMode;
+        mNextSourceAction = nextSourceAction;
+        mReceiptEmail = receiptEmail;
+        mShippingMethod = shippingMethod;
+        mSource = source;
+        mStatus = status;
+    }
+
+    public static String parseIdFromClientSecret(String clientSecret) {
+        return clientSecret.split("_secret")[0];
+    }
+
+    @Nullable
+    public static PaymentIntent fromString(@Nullable String jsonString) {
+        try {
+            return fromJson(new JSONObject(jsonString));
+        } catch (JSONException ignored) {
+            return null;
+        }
+    }
+
+    public static PaymentIntent fromJson(@Nullable JSONObject jsonObject) {
```

Comment:
Should this be annotated with `@Nullable`? Should the getters be annotated too?

On iOS I've made most of the fields optional, but do require `id`, `amount`, `clientSecret`, `currency`, `livemode`, and `status`

Reply:
added